public/app/routes/routes.tsx

```diff
@@ -540,3 +545,36 @@ export function getDynamicDashboardRoutes(cfg = config): RouteDescriptor[] {
     },
   ];
 }
+
+// Standalone plugin pages
+// These pages are registered by plugins and are showing up under different sections based on the Grafana configuration.
+function getStandalonePluginPageRoutes(): RouteDescriptor[] {
+  const state = store.getState();
+  const { navIndex } = state;
+  const isStandalonePage = (id: string) => id.includes('standalone-plugin-page-/');
+  const isDefaultAppRoute = (id: string) => id.includes('standalone-plugin-page-/a/'); // any page that is still served under "/a/:pluginId/*"
+  const standalonePageNavIds = Object.keys(navIndex).filter((id) => isStandalonePage(id) && !isDefaultAppRoute(id));
+
+  if (!standalonePageNavIds.length) {
+    return [];
+  }
+
+  return standalonePageNavIds.map((navId) => {
+    const { pluginId } = navIndex[navId];
+    const baseUrl = '';
+
+    return {
+      path: navIndex[navId].url || '',
+      exact: false,
+      component: ({ match, location, route, history, queryParams }) => (
+        <AppRootPage
+          route={route}
+          match={{ ...match, url: baseUrl, params: { ...match.params, pluginId } }}
+          queryParams={queryParams}
+          history={history}
+          location={location}
+        />
+      ),
+    };
+  });
```

Comment:
This feels a bit messy. 

Feels like we might have an opportunity here to simply logic. What if we registered all app plugin routes this way, and insted of passing pluginId via match it was a regular prop. And we could also then pass in the navId, so we would no longer need the messy logic getPluginSection ?

Reply:
I really like this idea from @torkelo! So basically clean up the "regular" routes and use the `navIndex` in state as source to dynamically generate the routes needed by app plugins. If I understood everything correct? Then you would be able to pick up the pluginId from the state instead of passing it via the match and it can easily be passed to `AppRootPage` as a prop.

Do we have any scenarios where we could end up with apps missing from the state due to some reason and hence not being registered? I'm thinking pre-loading of app plugins etc.

Reply:
yes, exactly. 

> Do we have any scenarios where we could end up with apps missing from the state due to some reason and hence not being registered? I'm thinking pre-loading of app plugins etc.

No all plugins are in the navIndex which is build on the backend based on plugin.json so does not need the plugin to be started.

Reply:
Very good idea, thanks @torkelo and @mckn! 👍  

I have changed the code accordingly, please have a look if this is close to what you were thinking about. There is still a lot of opportunity for cleaning up and refactoring around the `AppRootPage`, but I would not go further than this in this PR.

 **A few notes on the new changes:**
-  We are now only allowing plugins to render under URLs other than `/a/...` if they are specified as a standalone plugin page using the Grafana config. Like this plugins cannot override or extend core functionality without explicit configuration done by the server admin.
- `isCorePage` - The name of this new field was just a quick idea, we can change it if we find a better one. It is needed because otherwise the backend doesn't know if it should render the page under the plugin section in case it was not configured as a standalone-plugin-page using `[navigation.app_standalone_pages]`. (Plugins can register page includes with custom URLs that can appear as "links" under the plugins nav section, which is fine, however if a plugin is registering a page include to extend / override core functionality, then it should only appear and work if it is configured correctly in Grafana as well.)

Reply:
I have also updated the description of the PR to explain what is happening and to give an example on how to use this.